hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -927,7 +927,7 @@ def prescreen_buffer(self, buffer):
             except KeyError:
                 pass
             try:
-                b = min(b, self.capped[node_index])
+                b = b & self.capped[node_index]
```

Comment:
nit (local convention: "I'm being a pedant and it's up to you if you want to do anything about this rather than a requirement of review"): Now that you've fixed this to be a mask rather than a cap, I wonder if we should rename the `capped` field to `masks`? In the cases we use it it does actually function as both, but if we ever decide to put something other than a value of the form `2 ** n - 1` in there the name would be misleading.

Reply:
I won't change this right now, because `generate_novel_prefix` is still using it as a cap, so both names are “wrong”.

But this is a fine idea for a separate future PR.